### DISCUSSION TITLE
licensing issues on public domains

### DISCUSSION THREAD

**Discussion by alexgit2k** [craftcms/cms#14484] [Q&A] [answered]
Craft CMS 4.8 disables the control panel when licensing issues occur on public domains (see https://github.com/craftcms/cms/blob/4.8.0/CHANGELOG.md). How does Craft CMS check public domains?

We do have several test-/staging-domains which use subdomains of our main-domain, but are not public accessible, because they are either within our internal network (test) or password protected (staging).

**Comment by daltonrooney** [comment, marked as answer]
I think this is the KB article you're looking for: https://craftcms.com/knowledge-base/how-craft-license-enforcement-works#how-do-we-determine-craft-is-running-on-a-public-domain

> Does it have a dev-sounding subdomain?
> Exhaustively: acc, acceptance, ci, craftdemo, dev, integration, loc, local, preprod, preview, qa, sandbox, stage, staging, systest, test, testing, uat.

We usually include .dev or .staging in the subdomain.
